Backend/app/special-zone/src/Domain/Types/SpecialZone.hs

```diff
@@ -0,0 +1,155 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Types.SpecialZone where
+
+import Data.Aeson
+import Kernel.Prelude
+import Kernel.Types.Id
+import Kernel.Utils.JSON
+
+data SpecialZone = SpecialZone
+  { id :: Id SpecialZone,
+    name :: Text,
+    categoryCode :: Category,
+    geoJson :: ShapeFile,
+    city :: Text,
+    state :: Text,
+    createdAt :: UTCTime,
+    updatedAt :: UTCTime
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
+data SpecialZoneAPIEntity = SpecialZoneAPIEntity
+  { name :: Text,
+    categoryCode :: Category,
+    geoJson :: ShapeFile,
+    city :: Text,
+    state :: Text
+  }
+  deriving (Generic, Show, Eq, FromJSON, ToJSON, ToSchema)
+
+data EntryExit = EntryExit
```

Comment:
this type should be in `Domain.Types.EntryExit` module

Reply:
`EntryExit` uses `SpecialZoneId`, `SpecialZone` uses `EntryExit`, it will have cyclic dependency